app/src/main/java/com/example/sony/training/MainActivity.java

```diff
@@ -1,13 +1,295 @@
+    private TextView totalTextView;
+    private Spinner spinnerBill;
+    private EditText mEditAmount;
+    private EditText billAmountEditText;
+    private TextView percentTextView;
+    private SeekBar percentSeekBar;
+    private TextView tipTextView;
+    private RadioGroup roundingRadioGroup;
+    private RadioButton roundNoneRadioButton;
+    private RadioButton roundTipRadioButton;
+    private RadioButton roundTotalRadioButton;
+    private Spinner splitSpinner;
+    private TextView perPersonLabel;
+    private TextView perPersonTextView;
+
+    // xác định đối tượng SharedPreferences
+    private SharedPreferences savedValues;
+
+    // định nghĩa các hằng số làm tròn
+    private final int ROUND_NONE = 0;
+    private final int ROUND_TIP = 1;
+    private final int ROUND_TOTAL = 2;
+
+    // xác định các biến
+    private String billAmountString = "";
+    private float tipPercent = .15f;
+    private int rounding = ROUND_NONE;
+    private int split = 1;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        setContentView(R.layout.activity_main_widgets_bill);
+
+        // get references to the widgets
+        billAmountEditText = (EditText) findViewById(R.id.editAmount);
+        percentTextView = (TextView) findViewById(R.id.percentTextView);
+        percentSeekBar = (SeekBar) findViewById(R.id.seekBarPercent);
+        tipTextView = (TextView) findViewById(R.id.tipTextView);
+        totalTextView = (TextView) findViewById(R.id.totalTextView);
+        roundingRadioGroup = (RadioGroup)
+                findViewById(R.id.roundingRadioGroup);
+        roundNoneRadioButton = (RadioButton)
+                findViewById(R.id.radioBtnNone);
+        roundTipRadioButton = (RadioButton)
+                findViewById(R.id.radioBtnTip);
+        roundTotalRadioButton = (RadioButton)
+                findViewById(R.id.radioBtnTotal);
+        splitSpinner = (Spinner) findViewById(R.id.spinnerBill);
+        perPersonLabel = (TextView) findViewById(R.id.percentLabel);
+        perPersonTextView = (TextView) findViewById(R.id.percentTextView);
+
+        // set array adapter for spinner
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(
+                this, R.array.split_array, android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(
+                android.R.layout.simple_spinner_dropdown_item);
+        splitSpinner.setAdapter(adapter);
+
+        // set the listeners
+        billAmountEditText.setOnEditorActionListener(this);
+        billAmountEditText.setOnKeyListener(this);
+        percentSeekBar.setOnSeekBarChangeListener(this);
+        percentSeekBar.setOnKeyListener(this);
+        roundingRadioGroup.setOnCheckedChangeListener(this);
+        roundingRadioGroup.setOnKeyListener(this);
+        splitSpinner.setOnItemSelectedListener(this);
```

Comment:
pls define ``initEvents`` methods and move this code (line 91 -> 98)  into this method